perma_web/api2/views.py

```diff
@@ -0,0 +1,483 @@
+import django_filters
+from django.core.exceptions import ObjectDoesNotExist, ValidationError as DjangoValidationError
+from django.http import Http404
+from mptt.exceptions import InvalidMove
+from rest_framework import status
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework.filters import DjangoFilterBackend, SearchFilter, OrderingFilter
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from perma.utils import run_task
+from perma.tasks import upload_to_internet_archive, delete_from_internet_archive, run_next_capture
+from perma.models import Folder, CaptureJob, Link, Capture, Organization
+
+from utils import TastypiePagination, load_parent, raise_validation_error
+from serializers import FolderSerializer, CaptureJobSerializer, LinkSerializer, AuthenticatedLinkSerializer, \
+    LinkUserSerializer, OrganizationSerializer
+
+
+### BASE VIEW ###
+
+class BaseView(APIView):
+    permission_classes = (IsAuthenticated,)  # by default all users must be authenticated
+    serializer_class = None  # overridden for each subclass
+
+    # configure filtering of list endpoints by query string
+    filter_backends = (
+        DjangoFilterBackend,  # subclasses can be filtered by keyword if filter_class is set
+        SearchFilter,         # subclasses can be filtered by q= if search_fields is set
+        OrderingFilter        # subclasses can be ordered by order_by= if ordering_fields is set
+    )
+    ordering_fields = ()      # lock down order_by fields -- security risk if unlimited
+
+
+    ### helpers ###
+
+    def filter_queryset(self, queryset):
+        """
+        Given a queryset, filter it with whichever filter backend is in use.
+
+        Copied from GenericAPIView
+        """
+        try:
+            for backend in list(self.filter_backends):
+                queryset = backend().filter_queryset(self.request, queryset, self)
+            return queryset
+        except DjangoValidationError as e:
+            raise ValidationError(e.error_dict)
+
+    def get_object_for_user(self, user, queryset):
+        """
+            Get single object from queryset, making sure that returned object is accessible_to(user).
+        """
+        try:
+            obj = queryset.get()
+        except ObjectDoesNotExist:
+            raise Http404
```

Comment:
It's right not to handle MultipleObjectsReturned, right? If this function is misused that way, it SHOULD raise 500.